http.go

```diff
@@ -70,6 +70,10 @@ func (s *Server) ServeHTTPS() {
 		log.Fatalf("FATAL: loading tls config (%s, %s) failed - %s", s.Opts.TLSCertFile, s.Opts.TLSKeyFile, err)
 	}
 
+	if len(s.Opts.TLSClientCAFiles) > 0 {
+		config.ClientAuth = tls.RequestClientCert
```

Comment:
I'm fairly sure omitting the ClientCAs causes TLS not to verify.

Separate port wouldn't help because the router wouldn't expose it.